tests/test_irbempy.py

```diff
@@ -314,6 +396,52 @@ def test_find_footpoint(self):
         numpy.testing.assert_almost_equal(expected['Bfoot'], ans['Bfoot'], decimal=5)
         numpy.testing.assert_almost_equal(expected['loci'].data, ans['loci'].data, decimal=5)
 
+    def test_find_footpoint_bad_hemi(self):
+        with self.assertRaises(ValueError):
+            ib.find_footpoint(self.ticks, self.loci, hemi='asdf', omnivals=self.omnivals)
+
+    def test_get_Bfield_badval(self):
+        """test get_Bfield hitting badval"""
+        loci_noncredible = spacepy.coordinates.Coords([[-1e4, 1e4, 0], [2, 0, 0]], 'GEO', 'car', use_irbem=True)
+        actual = ib.get_Bfield(self.ticks, loci_noncredible, omnivals=self.omnivals)
+        self.assertTrue(np.isnan(actual["Blocal"][0]))
+        self.assertTrue(np.isnan(actual["Bvec"][0]).all())
+
+    def test_find_Bmirror_badval(self):
+        """test find_Bmirror hitting badval"""
+        loci_noncredible = spacepy.coordinates.Coords([[-1e4, 1e4, 0], [2, 0, 0]], 'GEO', 'car', use_irbem=True)
```

Comment:
Can we try this where the location is credible, but the pitch angle is in the loss cone? Here (if I'm reading this right) the PA 0 case is also the "way out in the Universe" case.

Reply:
Yes, I think the reason I used a non-credible location was because I couldn't make Blocal hit badval otherwise. It seems appropriate to me to split this test up into two cases: noncredible location with a reasonable pitch angle, and credible location with zero pitch angle.